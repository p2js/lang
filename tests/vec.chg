
type Vec<T> constructible(mkVec) = {
    ptr: T*,
    len: i32,
    cap: i32
};

function mkVec<T>(): Vec<T> {
    return .{ ptr: 0, len: 0, cap: 0 };
}

function main(): i32 {
    return 0;
}
