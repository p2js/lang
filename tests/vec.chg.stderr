

    return .{ ptr: 0, len: 0, cap: 0 };
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  type error: i32 is not a subtype of T* (reason: 'Return')
